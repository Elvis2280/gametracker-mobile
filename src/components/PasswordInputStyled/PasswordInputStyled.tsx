import React, { useState, type ReactElement } from 'react'
import { Input, View } from 'tamagui'
import { Ionicons } from '@expo/vector-icons'
import { Pressable } from 'react-native'

interface Props {
  onChange: (text: string) => void
  onBlur: () => void
  value: string
}

export default function PasswordInputStyled({
  onBlur,
  onChange,
  value
}: Props): ReactElement {
  const [isSecure, setIsSecure] = useState(true)
  const [textValue, setTextValue] = useState(value)

  const handleOnChange = (text: string): void => {
    setTextValue(text)
    onChange(text)
  }
  return (
    <View position="relative">
      <Input
        testID="password-input"
        onChangeText={handleOnChange}
        onBlur={onBlur}
        value={textValue}
        secureTextEntry={isSecure}
        placeholder="Password"
      />
      <View
        position="absolute"
        w="$4"
        display="flex"
        justifyContent="center"
        alignItems="center"
        right={10}
        top={'$2'}
      >
        <Pressable
          testID="password-input-toggle"
          onPress={() => {
            setIsSecure(!isSecure)
          }}
        >
          <Ionicons
            name={isSecure ? 'eye-off' : 'eye'}
            size={28}
            color={'lightgrey'}
          />
        </Pressable>
      </View>
    </View>
  )
}
