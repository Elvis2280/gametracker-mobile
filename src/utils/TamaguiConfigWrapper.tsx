import React, { type ReactElement } from 'react'
import { TamaguiProvider, Theme } from 'tamagui'
import tamaguiConfig from '../../tamagui.config'

interface Props {
  children: ReactElement
}

export default function TamaguiConfigWrapper({
  children
}: Props): ReactElement {
  return (
    <TamaguiProvider config={tamaguiConfig}>
      <Theme name={'blue_alt1'}>{children}</Theme>
    </TamaguiProvider>
  )
}
