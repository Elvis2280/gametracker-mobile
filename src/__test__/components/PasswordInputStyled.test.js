/* eslint-disable @typescript-eslint/comma-dangle */
/* eslint-disable @typescript-eslint/explicit-function-return-type */
import React from 'react'
import renderer from 'react-test-renderer'
import { render, fireEvent } from '@testing-library/react-native'
import PasswordInputStyled from '../../components/PasswordInputStyled/PasswordInputStyled'
import TamaguiConfigWrapper from '../../utils/TamaguiConfigWrapper'
import '@testing-library/jest-native/extend-expect'

describe('PasswordInputStyled', () => {
  it('renders correctly', () => {
    const tree = renderer
      .create(
        <TamaguiConfigWrapper>
          <PasswordInputStyled />
        </TamaguiConfigWrapper>
      )
      .toJSON()
    expect(tree).toMatchSnapshot()
  })

  it('renders correctly with props', () => {
    const handler = (value) => {}
    const value = 'test'
    const onblur = () => {}

    const rendered = renderer
      .create(
        <TamaguiConfigWrapper>
          <PasswordInputStyled
            value={value}
            onChange={handler}
            onBlur={onblur}
          />
        </TamaguiConfigWrapper>
      )
      .toJSON()

    expect(rendered).toMatchSnapshot()
  })

  it('renders correctly value from props and change on user change text', () => {
    const value = 'test'
    const handler = (value) => {}
    const { getByTestId } = render(
      <TamaguiConfigWrapper>
        <PasswordInputStyled value={value} onChange={handler} />
      </TamaguiConfigWrapper>
    )

    const input = getByTestId('password-input')
    expect(input.props.value).toBe('test')

    fireEvent.changeText(input, 'new value')
    expect(input.props.value).toBe('new value')
  })

  it('should change the secureTextEntry prop when the eye icon is pressed', () => {
    const { getByTestId } = render(
      <TamaguiConfigWrapper>
        <PasswordInputStyled />
      </TamaguiConfigWrapper>
    )

    const input = getByTestId('password-input')
    const eyeIcon = getByTestId('password-input-toggle')

    expect(input.props.secureTextEntry).toBe(true)

    fireEvent.press(eyeIcon)
    expect(input.props.secureTextEntry).toBe(false)

    fireEvent.press(eyeIcon)
    expect(input.props.secureTextEntry).toBe(true)
  })
})
