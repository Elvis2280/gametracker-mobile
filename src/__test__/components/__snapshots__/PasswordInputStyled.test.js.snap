// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`PasswordInputStyled renders correctly 1`] = `
<View
  style={
    {
      "position": "relative",
    }
  }
>
  <TextInput
    focusable={true}
    onChangeText={[Function]}
    outlineWidth={0}
    placeholder="Password"
    placeholderTextColor="hsl(206, 100%, 50.0%)"
    secureTextEntry={true}
    style={
      {
        "backgroundColor": "hsl(214, 59.4%, 15.3%)",
        "borderBottomColor": "hsl(206, 100%, 50.0%)",
        "borderBottomLeftRadius": 9,
        "borderBottomRightRadius": 9,
        "borderBottomWidth": 1,
        "borderLeftColor": "hsl(206, 100%, 50.0%)",
        "borderLeftWidth": 1,
        "borderRightColor": "hsl(206, 100%, 50.0%)",
        "borderRightWidth": 1,
        "borderStyle": "solid",
        "borderTopColor": "hsl(206, 100%, 50.0%)",
        "borderTopLeftRadius": 9,
        "borderTopRightRadius": 9,
        "borderTopWidth": 1,
        "color": "hsl(206, 98.0%, 95.8%)",
        "fontFamily": "Inter",
        "fontSize": 14,
        "height": 44,
        "letterSpacing": 0,
        "minWidth": 0,
        "paddingLeft": 16,
        "paddingRight": 16,
      }
    }
    testID="password-input"
  />
  <View
    style={
      {
        "alignItems": "center",
        "display": "flex",
        "justifyContent": "center",
        "position": "absolute",
        "right": 10,
        "top": 7,
        "width": 44,
      }
    }
  >
    <Text />
  </View>
</View>
`;

exports[`PasswordInputStyled renders correctly with props 1`] = `
<View
  style={
    {
      "position": "relative",
    }
  }
>
  <TextInput
    focusable={true}
    onChangeText={[Function]}
    outlineWidth={0}
    placeholder="Password"
    placeholderTextColor="hsl(206, 100%, 50.0%)"
    secureTextEntry={true}
    style={
      {
        "backgroundColor": "hsl(214, 59.4%, 15.3%)",
        "borderBottomColor": "hsl(206, 100%, 50.0%)",
        "borderBottomLeftRadius": 9,
        "borderBottomRightRadius": 9,
        "borderBottomWidth": 1,
        "borderLeftColor": "hsl(206, 100%, 50.0%)",
        "borderLeftWidth": 1,
        "borderRightColor": "hsl(206, 100%, 50.0%)",
        "borderRightWidth": 1,
        "borderStyle": "solid",
        "borderTopColor": "hsl(206, 100%, 50.0%)",
        "borderTopLeftRadius": 9,
        "borderTopRightRadius": 9,
        "borderTopWidth": 1,
        "color": "hsl(206, 98.0%, 95.8%)",
        "fontFamily": "Inter",
        "fontSize": 14,
        "height": 44,
        "letterSpacing": 0,
        "minWidth": 0,
        "paddingLeft": 16,
        "paddingRight": 16,
      }
    }
    testID="password-input"
    value="test"
  />
  <View
    style={
      {
        "alignItems": "center",
        "display": "flex",
        "justifyContent": "center",
        "position": "absolute",
        "right": 10,
        "top": 7,
        "width": 44,
      }
    }
  >
    <Text
      allowFontScaling={false}
      onPress={[Function]}
      selectable={false}
      style={
        [
          {
            "color": "lightgrey",
            "fontSize": 28,
          },
          undefined,
          {
            "fontFamily": "ionicons",
            "fontStyle": "normal",
            "fontWeight": "normal",
          },
          {},
        ]
      }
    >
      ïŠ´
    </Text>
  </View>
</View>
`;
